"tox" run failing tests on Windows

I am a proud user of a VM with Windows 7 and conda + pip installed dandi development environment...  Running `tox` resulted in:

```
====================================================================================== short test summary info =======================================================================================
FAILED dandi/support/tests/test_cache.py::test_memoize_path - OSError: [WinError 1314] A required privilege is not held by the client: 'file.dat' -> 'C:\\Users\\DataLad\\AppData\\Local\\Temp\\pyte...

FAILED dandi/support/tests/test_iterators.py::test_IteratorWithAggregation - assert False
FAILED dandi/tests/test_organize.py::test_organize_nwb_test_data[simulate] - assert 1 == 0
FAILED dandi/tests/test_organize.py::test_organize_nwb_test_data[symlink] - assert 1 == 0
FAILED dandi/tests/test_organize.py::test_organize_nwb_test_data[auto] - assert False
FAILED dandi/tests/test_organize.py::test_detect_link_type[True-True-symlink] - AssertionError: assert 'hardlink' == 'symlink'
FAILED dandi/tests/test_organize.py::test_detect_link_type[True-False-symlink] - AssertionError: assert 'copy' == 'symlink'
============================================================================== 7 failed, 56 passed, 9 skipped in 39.81s ==============================================================================
Exception ignored in: <ssl.SSLSocket fd=460, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('10.0.2.15', 51095), raddr=('3.19.164.171', 443)>
ResourceWarning: unclosed <ssl.SSLSocket fd=460, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('10.0.2.15', 51095), raddr=('3.19.164.171', 443)>
ERROR: InvocationError for command 'C:\Users\DataLad\PycharmProjects\dandi-cli\.tox\py3\Scripts\coverage.EXE' run -m pytest -v dandi (exited with code 1)
______________________________________________________________________________________________ summary _______________________________________________________________________________________________
  lint: commands succeeded
ERROR:   py3: commands failed

```

not sure why does not fail for us on Windows github CI runs.

--- a/dandi/support/tests/test_cache.py
+++ b/dandi/support/tests/test_cache.py
@@ -142,7 +142,10 @@ def test_memoize_path(cache, tmp_path):
 
     # Check that symlinks should be dereferenced
     symlink1 = str(tmp_path / (fname + ".link1"))
-    os.symlink(fname, symlink1)
+    try:
+        os.symlink(fname, symlink1)
+    except OSError:
+        pass
     if op.islink(fname):  # hopefully would just skip Windows if not supported
         ncalls = len(calls)
         assert memoread(symlink1, 0) == "Content"

--- a/dandi/tests/test_organize.py
+++ b/dandi/tests/test_organize.py
@@ -1,6 +1,7 @@
 import os
 from glob import glob
 import os.path as op
+from pathlib import Path
 import ruamel.yaml
 
 from ..consts import file_operation_modes
@@ -115,6 +116,19 @@ def test_organize_nwb_test_data(nwb_test_data, tmpdir, clirunner, mode):
         nwb_test_data = op.relpath(nwb_test_data, cwd)
         outdir = op.relpath(outdir, cwd)
 
+    src = Path(tmpdir, "src")
+    src.touch()
+    dest = Path(tmpdir, "dest")
+    try:
+        dest.symlink_to(src)
+    except OSError:
+        symlinks_work = False
+    else:
+        symlinks_work = True
+
+    if mode in ("simulate", "symlink") and not symlinks_work:
+        pytest.skip("Symlinks not supported")
+
     input_files = op.join(nwb_test_data, "v2.0.1")
 
     cmd = ["-d", outdir, "--files-mode", mode, input_files]
@@ -146,7 +160,7 @@ def test_organize_nwb_test_data(nwb_test_data, tmpdir, clirunner, mode):
 
     if mode == "simulate":
         assert all((op.isabs(p) != relative) for p in produced_paths)
-    elif mode == "symlink" or mode == "auto":
+    elif mode == "symlink" or (mode == "auto" and symlinks_work):
         assert all(op.islink(p) for p in produced_nwb_paths)
     else:
         assert not any(op.islink(p) for p in produced_paths)
@@ -216,11 +230,12 @@ def test_ambiguos_probe1():
     ],
 )
 def test_detect_link_type(monkeypatch, tmp_path, sym_success, hard_success, result):
+    def succeed_link(src, dest):
+        pass
+
     def error_link(src, dest):
         raise OSError("Operation failed")
 
-    if not sym_success:
-        monkeypatch.setattr(os, "symlink", error_link)
-    if not hard_success:
-        monkeypatch.setattr(os, "link", error_link)
+    monkeypatch.setattr(os, "symlink", succeed_link if sym_success else error_link)
+    monkeypatch.setattr(os, "link", succeed_link if hard_success else error_link)
     assert detect_link_type(tmp_path) == result

--- a/dandi/tests/test_organize.py
+++ b/dandi/tests/test_organize.py
@@ -125,9 +125,21 @@ def test_organize_nwb_test_data(nwb_test_data, tmpdir, clirunner, mode):
         symlinks_work = False
     else:
         symlinks_work = True
+    try:
+        dest.unlink()
+    except FileNotFoundError:
+        pass
+    try:
+        os.link(src, dest)
+    except OSError:
+        hardlinks_work = False
+    else:
+        hardlinks_work = True
 
     if mode in ("simulate", "symlink") and not symlinks_work:
         pytest.skip("Symlinks not supported")
+    elif mode == "hardlink" and not hardlinks_work:
+        pytest.skip("Hard links not supported")
 
     input_files = op.join(nwb_test_data, "v2.0.1")
 

